src/services/jsTyping.ts

```diff
@@ -6,40 +6,81 @@
+
     type HostType = {
+        directoryExists: (fileName: string) => boolean;
         fileExists: (fileName: string) => boolean;
         readFile: (path: string, encoding?: string) => string;
         readDirectory: (path: string, extension?: string, exclude?: string[], depth?: number) => string[];
         writeFile: (path: string, data: string, writeByteOrderMark?: boolean) => void;
     };
 
-    interface TsdJsonCacheInfo {
-        path: string;
-        value: any;
-    }
-
     var _host: HostType;
+
+    var autoTypingsComment =
+        `//  This file contains a list of JavaScript typings (d.ts) that have
+//  been auto-referenced to provide a better completion list experience.
+//  For more information see the following page on the TypeScript wiki:
+//  https://github.com/Microsoft/TypeScript/wiki/AutoTypings
+`;
+
+    var autoTypingsDisabled = `The autoTypingsEnabled property is disabled for this project, no typings are being auto-referenced.
+`;
     
     // a typing name to typing file path mapping
     var inferredTypings: Map<string> = {};
 
-    function tryParseJson(jsonPath: string) {
-        try {
-            return JSON.parse(jsonPath);
+    function tryParseJson(host: HostType, jsonPath: string): any {
+        if (host.fileExists(jsonPath)) {
+            try {
+                // Remove whitespace and strip out single-line comments
+                let contents = host.readFile(jsonPath).replace(/\/\/(.*)(\r*)(\n*)|\r?\n|\r|/g, "");
+                return JSON.parse(contents);
+            }
+            catch (e) { }
         }
-        catch (e) { return undefined; }
+        return undefined;
     }
     
     /**
      * @param cachePath is the path to the cache location, which contains a tsd.json file and a typings folder
      */
     export function discoverTypings(
-        host: HostType, fileNames: string[], cachePath: string, compilerOptions?: CompilerOptions, safeList?: string[], noDevDependencies?: boolean)
+        host: HostType, fileNames: string[], cachePath: string, typingsConfigPath: string, compilerOptions?: CompilerOptions, safeList?: string[], noDevDependencies?: boolean)
         : { cachedTypingPaths: string[], newTypingNames: string[] } {
         _host = host;
+
+        // Clear inferred typings map
+        inferredTypings = {};
+
         // Directories to search for package.json, bower.json and other typing information
         let searchDirs: string[] = [];
 
+        // Check the typings config path to see if auto-typings is
+        // enabled and if the there are any typings in the include/exclude list
+        let exclude: string[] = [];
+        let autoTypingsJsonPath = ts.combinePaths(ts.normalizePath(typingsConfigPath), "autoTypings.json");
+        let autoTypingsJsonDict = tryParseJson(host, autoTypingsJsonPath);
+        if (autoTypingsJsonDict) {
+            if (autoTypingsJsonDict.hasOwnProperty("autoTypingsEnabled") && !autoTypingsJsonDict["autoTypingsEnabled"]) {

```

Comment:
Also this seems to ignore `include` and `exclude` if `autoTypingEnabled` is false, is that what we want?